src/PrestaShopBundle/Bridge/Helper/Listing/HelperBridge/FeatureHelperListBridge.php

```diff
@@ -37,24 +37,39 @@
  */
 class FeatureHelperListBridge extends HelperListBridge
 {
+    /**
+     * {@inheritDoc}
+     */
     public function generateListQuery(
         HelperListConfiguration $helperListConfiguration,
         int $idLang
-    ): void {
-        parent::generateListQuery($helperListConfiguration, $idLang);
-
-        $nbItems = count($helperListConfiguration->list);
-        for ($i = 0; $i < $nbItems; ++$i) {
-            $item = &$helperListConfiguration->list[$i];
-
-            $query = new DbQuery();
-            $query->select('COUNT(fv.id_feature_value) as count_values');
-            $query->from('feature_value', 'fv');
-            $query->where('fv.id_feature =' . (int) $item['id_feature']);
-            $query->where('(fv.custom=0 OR fv.custom IS NULL)');
-            $res = Db::getInstance(_PS_USE_SQL_SLAVE_)->getValue($query);
-            $item['value'] = (int) $res;
-            unset($query);
+    ): string {
+        $listSql = parent::generateListQuery($helperListConfiguration, $idLang);
+
+        // adds feature_value count to evey row of feature
+        foreach ($helperListConfiguration->list as &$featureRowRecord) {
+            $this->addFeatureValuesCount($featureRowRecord);
```

Comment:
Consider doing this instead:
```suggestion
            $featureRowRecord['value'] = (int) $this->getFeatureValuesCount($featureRowRecord['id_feature']);
```

The code will be clearer and you won't need to pass along the array as a reference anymore